packages/terra-date-picker/src/DateUtil.jsx

```diff
@@ -142,6 +143,103 @@ class DateUtil {
     return undefined;
   }
 
+  /**
+   * Gets the date format order from moment's long date format.
+   * @param {string} momentDateFormat - Moment's long date format.
+   * @return {string} - The date format order. Possible Values (MDY, DMY, YMD)
+   */
+  static getDateFormatOrder(momentDateFormat) {
+    let dateFormatOrder;
+    if (momentDateFormat === 'MM/DD/YYYY') {
+      dateFormatOrder = DateUtil.dateOrder.MDY;
+    } else if (momentDateFormat === 'DD/MM/YYYY') {
+      dateFormatOrder = DateUtil.dateOrder.DMY;
+    } else if (momentDateFormat === 'YYYY/MM/DD') {
+      dateFormatOrder = DateUtil.dateOrder.YMD;
```

Comment:
imho we could do direct returns in each conditional, instead of saving the formats in a variable, then finally returning.
```suggestion
      return DateUtil.dateOrder.YMD;
```

Reply:
I tried this approach but am getting the following errors from the liner:

- Unnecessary else after return.
- Expected to return a value at the end of static method

Reply:
I think Derek's suggestion is to do something like this.

```
static getDateFormatOrder(momentDateFormat) {
    if (momentDateFormat === 'MM/DD/YYYY') {
      return DateUtil.dateOrder.MDY;
    } else if (momentDateFormat === 'DD/MM/YYYY') {
      return DateUtil.dateOrder.DMY;
    } else if (momentDateFormat === 'YYYY/MM/DD') {
      return DateUtil.dateOrder.YMD;
    }

    return '';  //Or should this return null/undefined?
  }
```

Reply:
Yeah that is exactly what i tried:
```
if (momentDateFormat === 'MM/DD/YYYY') {
    return DateUtil.dateOrder.MDY;
} else if (momentDateFormat === 'DD/MM/YYYY' || momentDateFormat === 'DD.MM.YYYY' || momentDateFormat === 'DD-MM-YYYY') {
    return DateUtil.dateOrder.DMY;
} else if (momentDateFormat === 'YYYY-MM-DD') {
    return DateUtil.dateOrder.YMD;
}

return '';
```
But, am getting the following linter error when implementing it as above:
<img width="629" alt="Screenshot 2020-11-04 at 5 25 39 PM" src="https://user-images.githubusercontent.com/50138162/98108965-d1f64b00-1ec2-11eb-9bfb-fba8e9199e47.png">

Reply:
Hmm. looks like it satisfies https://eslint.org/docs/rules/no-else-return. I guess just leave it the way it is.